src/lib.rs

```diff
@@ -142,6 +168,11 @@ impl Connection {
     }
 }
 
+// Extract the message type from the first 3 bits of the passed (header) byte
+fn get_message_type(byte: u8) -> u8 {
+    return (byte >> 5) & 0x7;
+}
```

Comment:
To simplify the code in `Connection::write`:
```suggestion
fn get_message_type(byte: u8) -> MessageType {
    MessageType::try_from((byte >> 5) & 0x7).unwrap_or(MessageType::Control)
}
```